### PR TITLE
chore(rust): make secretkey and publickey more usable

### PR DESCRIPTION
**Pull request by SanjoDeundiak**

### CODE REVIEW

File: implementations/rust/kex/src/x3dh.rs [hunk 1]
```diff
@@ -413,9 +414,9 @@ impl KeyExchanger for X3dhInitiator {
 
                 // Check the prekey_bundle signature
                 vault.verify(
-                    prekey_bundle.signature_prekey.0,
-                    prekey_bundle.identity_key.0.as_slice(),
-                    prekey_bundle.signed_prekey.0.as_slice(),
+                    prekey_bundle.signature_prekey.as_ref().clone(), /* FIXME */
```
Comment (SanjoDeundiak) [inline implementations/rust/kex/src/x3dh.rs:417]: @mikelodder7 should verify take array reference instead of array?

Comment (mikelodder7) [reply implementations/rust/kex/src/x3dh.rs:417]: yes it should

**Review by mikelodder7 — CHANGES_REQUESTED** [review]
just make the one change and it should be good.

File: implementations/rust/kex/src/x3dh.rs (hunk 1 above)
Comment (SanjoDeundiak) [reply implementations/rust/kex/src/x3dh.rs:417]: @mikelodder7 addressed in #704

**Review by mikelodder7 — APPROVED** [review]

**Pull request merged**